templates/nextflow-ecs-cluster.yaml

```diff
@@ -116,6 +116,40 @@ Resources:
       LogGroupName: !Join ['-', [ECSLogGroup, !Ref 'AWS::StackName']]
       RetentionInDays: 14
 
+  EcsLaunchTemplate:
+    Type: AWS::AutoScaling::LaunchTemplate
+    Properties:
+      ImageId: !Ref EcsAmiId
+      InstanceType: !Ref EcsInstanceType
+      IamInstanceProfile: !Ref EcsInstanceProfile
+      SecurityGroups:
+        - !Ref EcsSecurityGroupId
+      BlockDeviceMappings:
+        - DeviceName: !Ref RootDeviceName
+          Ebs:
+            VolumeSize: !Ref RootEbsVolumeSize
+            VolumeType: !Ref EbsVolumeType
+      UserData:
+        Fn::Base64: !Sub |
+          MIME-Version: 1.0
+          Content-Type: multipart/mixed; boundary="==BOUNDARY=="
+
+          --==BOUNDARY==
+          Content-Type: text/cloud-config; charset="us-ascii"
+
+          #cloud-config
+          repo_update: true
+          repo_upgrade: security
+
+          write_files:
+          - path: /etc/ecs/ecs.config
+            content: |
+              ECS_CLUSTER=${EcsClusterName}
+              ECS_IMAGE_PULL_BEHAVIOR=prefer-cached
+            append: true
+
+          --==BOUNDARY==--
```

Comment:
When I tested this, I ran into a failure. I think you meant to use `AWS::EC2::LaunchTemplate` as the type as opposed to `AWS::AutoScaling::LaunchTemplate`. After making that change, I had to nest the existing properties under `LaunchTemplateData`. I also had to add the `Arn` key under `IamInstanceProfile` as well as switch from `SecurityGroups` to `SecurityGroupIds`. Here's the version that I was able to deploy. Could you sanity-check it? 

```suggestion
Type: AWS::EC2::LaunchTemplate
    Properties:
      LaunchTemplateData:
        ImageId: !Ref EcsAmiId
        InstanceType: !Ref EcsInstanceType
        IamInstanceProfile:
          Arn: !GetAtt EcsInstanceProfile.Arn
        SecurityGroupIds:
          - !Ref EcsSecurityGroupId
        BlockDeviceMappings:
          - DeviceName: !Ref RootDeviceName
            Ebs:
              VolumeSize: !Ref RootEbsVolumeSize
              VolumeType: !Ref EbsVolumeType
        UserData:
          Fn::Base64: !Sub |
            MIME-Version: 1.0
            Content-Type: multipart/mixed; boundary="==BOUNDARY=="

            --==BOUNDARY==
            Content-Type: text/cloud-config; charset="us-ascii"

            #cloud-config
            repo_update: true
            repo_upgrade: security

            write_files:
            - path: /etc/ecs/ecs.config
              content: |
                ECS_CLUSTER=${EcsClusterName}
                ECS_IMAGE_PULL_BEHAVIOR=prefer-cached
              append: true

            --==BOUNDARY==--
```